### PR TITLE
OSS Documentation

### PR DESCRIPTION
**Pull request by smk0033**
OSS ticket here: https://github.com/NASA-IMPACT/active-maap-sprint/issues/746

So far:
- created new directory under "System Reference Guide" for the Docs site for the OSS documentation
- general oss doc
- document on data sharing/doi minting/licensing

### CODE REVIEW

**Comment by smk0033** [comment]
I'm sure these will need some more work and additions, but figured I'd push what I have so far for feedback

File: docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb
Comment (jjfrench) [inline docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>Maybe convert this to a list for readability?</p><p> </p><p>Software should be made available in a "publicly accessible repository that is widely recognized by the community", should be "citable using a persistent identifier" (i.e. a DOI), and "should be released under a permissive license that has broad acceptance in the community".</p><p> </p><p>Is there a link for "policy documentation"?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (jjfrench) [inline docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>Can we clarify which community for "already used within the community"</p><p> </p><p>Maybe add an asterisk to the bolded: <strong>If you are a MAAP user developing code, we recommend submitting your repository to the </strong><a href="https://zenodo.org/communities/maap?q=&amp;l=list&amp;p=1&amp;s=10&amp;sort=newest" target="_blank"><strong>MAAP Community on Zenodo</strong></a><strong>.</strong> to help distinguish it from a headed?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (jjfrench) [inline docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>Should we include a small statement about why you would choose one over the other?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='4'/>

File: docs/source/system_reference_guide/oss_documentation/general_oss.ipynb
Comment (jjfrench) [inline docs/source/system_reference_guide/oss_documentation/general_oss.ipynb:1]: <p>I think the second header ended up in this cell, should we move is down one?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

File: docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb
Comment (smk0033) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>Made into a list and mentioned to refer to the SPD-41a. There are also some additional policies - should those be included? I added to the disclaimer and said there can be additional policies if NASA-funded, but figured licensing/doi were the main ones we talked about for MAAP users.</p>

Comment (smk0033) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>Like Apache 2.0 over MIT or something like that? Or just why those three in general are good options?</p>

Comment (smk0033) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: Oops I missed the very first part of that comment when I made some adjustments. I'd have to look back at documentation, but I think it was just generally the Earth Science and OSS communities🤔 I can go back and add something like that though once I confirm!

Comment (smk0033) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: Or maybe mention all requirements for software, and have a section below that states if you're a MAAP user that isn't NASA-funded, we encourage the repository/doi/license?

Comment (jjfrench) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: this is what a chatbot came up with:

> MIT License:
> 
> It’s a short and simple permissive license.
> It allows you to use, copy, modify, merge, publish, distribute, sublicense, and sell copies of the software without many restrictions.
> It’s most similar to the 2-clause BSD license.
> 
> BSD 3-Clause License:
> 
> It’s similar to the MIT license but adds a term that prevents someone from claiming false endorsement.
> This non-endorsement clause prohibits promotion of any derived work using the name of the license or its authors.
> It’s compatible with GPLv2 and v3.
> 
> Apache 2.0 License:
> 
> It requires you to keep the license file, the NOTICE file if there is one, and show notice for modified files.
> It addresses some patent-related issues, so companies use it a lot.
> It’s compatible with GPLv3 but not v2 (due to the patent clauses).
> If developers desire the assurance of patent protection, Apache 2.0 is the better choice.



but this is really just a nice to have since users can click on the license links and read more

Comment (smk0033) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: Cool, that helps! I'll look into adding short statements about each one, thanks!

Comment (jjfrench) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>I think the SPD-41a link is sufficient if that's the one we're referring to</p>

Comment (wildintellect) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>I would send people to <a href="https://choosealicense.com" target="_blank">https://choosealicense.com/</a>  and  <a href="https://opensource.org" target="_blank">https://opensource.org</a> </p><p>I'm not sure we can make any statements on the specifics of the licenses.</p>

Comment (wildintellect) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p><a href="https://choosealicense.com/licenses" target="_blank">https://choosealicense.com/licenses/</a> is a good page. Note Choosealicense is MIT biased.</p>

File: docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb
Comment (wildintellect) [inline docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>The title says "Data" but everything on the page is about "Code". Change the name until a time when we details data sharing.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (wildintellect) [inline docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>data =&gt; code as mentioned above</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (wildintellect) [inline docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>data =&gt; code</p><p> </p><p>The USGS page doesn't seem to have a list of data repositories, this is just confusion in the sentence structure, reorder to put the link last after explaining it's purpose.</p><p> </p><p>"If you are NASA-Funded: " mixes in Data, which I don't think we are ready to address yet. Nothing else in the Document covers data yet.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

**Comment by wildintellect** [comment]
Overall this first version only covers Software/Code.

There's one paragraph about data and DAACs that's very incomplete. We can aim to put that in the next round of edits.
Core to Data is requirements of NASA funding, options for data storage (I think I made a list in the ticket), and what to do when a DAAC will not take your data.

**Comment by smk0033** [comment]
> Overall this first version only covers Software/Code.
> 
> There's one paragraph about data and DAACs that's very incomplete. We can aim to put that in the next round of edits. Core to Data is requirements of NASA funding, options for data storage (I think I made a list in the ticket), and what to do when a DAAC will not take your data.

Is next round of edits after we merge this first version? If so, I'll go ahead and remove the sentence about DAACs (unless the whole NASA-funded paragraph should be removed for now)

File: docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb
Comment (freitagb) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>Given that I don't know the difference it may be that other scientists don't and it would be a good addition</p>

Comment (freitagb) [inline docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>This is slightly confusing for me. The first two sentences talk about data, but the rest is software. Is that intentional? I think we'll likely have different guidelines for data given that we won't want data dois to come from Zenodo. We'll be working with ESDIS on the best path forward on data governance in VEDA and can bring that back to the documentation in MAAP.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

File: docs/source/system_reference_guide/oss_documentation/general_oss.ipynb
Comment (freitagb) [inline docs/source/system_reference_guide/oss_documentation/general_oss.ipynb:1]: <p>I would change data to content in the bulleted list here.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

File: docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb
Comment (smk0033) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>Oops, forgot to remove "data" from that first sentence. That sounds like a plan though, I think Alex also said we aren't ready just yet to talk about data</p>

Comment (omshinde) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>I agree to all the above comments.. It would be good to have a brief description or table highlighting usage of each license based on a few attributes such as - redistribution, commercial usage, sharing...</p>

File: docs/source/system_reference_guide/oss_documentation/general_oss.ipynb
Comment (omshinde) [inline docs/source/system_reference_guide/oss_documentation/general_oss.ipynb:1]: <p>&gt;...provides guidance on how NASA-funded projects and activities should share publications, data, and software</p><p>...</p><p> </p><p>I was thinking if it would be good to provide one example for each of these. It's always easier to understand from examples. However, since we have already added the original source please feel free to ignore it. </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

**Comment by omshinde** [comment]
Thank you for your efforts @smk0033 ..This looks great. I have finished my review.

**Comment by smk0033** [comment]
Removed sentence about DAAC submission until next edits

**Review by omshinde — APPROVED** [review]

File: docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb
Comment (wildintellect) [inline docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>Should this be "Open Sourcing Code" subtitle can be the current text. (not a blocker just an idea).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

Comment (wildintellect) [inline docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>We should also be linking to TOPS <a href="https://nasa.github.io/Transform-to-Open-Science" target="_blank">https://nasa.github.io/Transform-to-Open-Science/</a></p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='1'/>

Comment (wildintellect) [inline docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>SMD (Science Mission Directorate) &lt;- Science Mission Directorate (SMD)</p><p>We don't call out uploading code to a public repository. Github, MAAP Gitlab or directly to Zenodo are acceptable (other public long term code repositories could also be acceptable, this is just a list of common options).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='2'/>

Comment (wildintellect) [inline docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>version controlled platform &lt;- version control platform</p><p> </p><p>How did we pick the list? I think I made a list of other common options in another ticket. University affiliated repositories is also acceptable (e.g. Dataverse etc).</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='3'/>

Comment (smk0033) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>I think I pulled them from page 16 of the Open-Source Science Guidance doc under their "Where to Share Software" section, but I'll look back at your list and add more! They have some other generalist repos also listed on page 12 for where to share data, I think Dataverse is listed there</p>

Comment (jjfrench) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>Maybe instead of "... made available in a publicly accessible repository that is widely recognized by the community." we could say "... made available in publicly accessible repositories such as Github, MAAP Gitlab or Zenodo." ?</p><p> </p><p> </p>

Comment (smk0033) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: Hmm. The only thing is I pulled these quotes and information directly from the SPD-41a, and this section is specifically stating overall guidelines for NASA-funded projects. I could change that section though. Would it work if I had a statement under the repositories section (under not NASA-funded) stating that Github and MAAP Gitlab are also acceptable for MAAP users? I do have Zenodo down there and the statement to submit a repo to the Zenodo MAAP Community

Comment (jjfrench) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>Oh, good point! Could just update that list to include those other public repositories. </p><p> </p><p>I wonder if you need the "If you are not NASA-Funded:" subheading then if some are applicable to both types of projects</p><p> </p><p>Thoughts?</p>

Comment (wildintellect) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: overall, anything that works for NASA funded is also true for non-NASA funded, with the exception of NASA only resources. So the venn diagram is highly overlapping with a few NASA only options, mostly shared options, and then other reputable open science places for non-NASA work.

Comment (smk0033) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: 👍 went ahead and removed the "If you are not NASA-Funded:" subheader, added some extra repos, and then made a note about Github/Gitlab for MAAP users

File: docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb
Comment (jjfrench) [inline docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: <p>This is kind of picky, if we put MAAP Gitlab under notes for MAAP users, I think we should link the MAAP Github as well. </p><p> </p><p>The generic Github link might go under the first bulleted list and generic Gitlab under the other well-known repositories.</p><p> </p><p>What do you think? </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/MAAP-Project/maap-documentation/pull/357/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='5'/>

Comment (smk0033) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: That's a good idea! Before I adjust the links and wording under notes for MAAP users, @wildintellect users are allowed to create repos within the MAAP Gitlab and MAAP Github?

Comment (wildintellect) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: MAAP users do not have permissions to add repos to github.com/maap-project this is very different from the MAAP gitlab. If using GitHub it would be recommended that users create a new Org for their research project.

Comment (wildintellect) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: Gitlab yes
Github no, that has to be under their own Org or User (Org recommended for collaborative science). the MAAP-Project Org is reserved for "official" uses, users can contribute to those repos like any other open source project.

Comment (smk0033) [reply docs/source/system_reference_guide/oss_documentation/doi_and_licensing.ipynb:1]: Thanks! Edited to adjust the Github/Gitlab stuff, let me know if anything need to be adjusted.

**Review by wildintellect — APPROVED** [review]
This is a great first version to publish. I'm sure there will be revisions based on UWG feedback.

**Pull request merged**
